File: alternating_sum.py
def alternatingSum(nums):
    total = 0
    for i in range(len(nums)):
        if i % 2 == 0:
            total += nums[i]  # 짝수 인덱스 더하기
        else:
            total -= nums[i]  # 홀수 인덱스 빼기
    return total

# 예시 실행
print(alternatingSum([1, 3, 5, 7]))  # -4
print(alternatingSum([100]))         # 100

File: alternating_sum.cpp
#include <iostream>
#include <vector>
using namespace std;

int alternatingSum(vector<int>& nums) {
    int sum = 0;
    for (int i = 0; i < nums.size(); i++) {
        if (i % 2 == 0) {
            sum += nums[i];
        }
        else {
            sum -= nums[i];
        }
    }
    return sum;
}

int main() {
    vector<int> nums0 = {1, 3, 5, 7};
    vector<int> nums1 = {100};
    cout << alternatingSum(nums0) << '\n';
    cout << alternatingSum(nums1) << '\n';
    return 0;
}
